### ISSUE TITLE
[New Rule] Check propTypes are static

### ISSUE THREAD

**Issue by pke** [yannickcr/eslint-plugin-react#1896]
You get this runtime warning:

`warning: propTypes was defined as an instance property on MyComponent. Use a static property to define propTypes instead.`

when you forget to make the `propTypes` static like this:

```jsx
class MyComponent extends React.PureComponent {
  propTypes = {
    text: PropTypes.string,
  }
}
```

The new rule could detect non-static `propTypes` inside class definitions and report them by default as warning.

**Comment by ljharb** [comment]
This seems like something `no-typos` should be handling.

**Comment by ljharb** [comment]
Indeed, no-typos already handled this over a year before this was filed (see #1209)
